net-p2p/gridcoin/gridcoin-5.3.3.0.ebuild

```diff
@@ -0,0 +1,164 @@
+
+}
+
+src_prepare() {
+	if use debug && [[ ! $(portageq envvar FEATURES) =~ .*(splitdebug|nostrip).* ]]; then
+		ewarn "You have enabled debug flags and macros during compilation."
+		ewarn "For these to be useful, you should also have Portage retain debug symbols."
+		ewarn "See https://wiki.gentoo.org/wiki/Debugging on configuring your environment"
+		ewarn "and set your desired FEATURES before (re-)building this package."
+	fi
+	default
+	./autogen.sh
+}
+
+src_configure() {
+	use harden && append-flags -Wa,--noexecstack
+	econf \
+		$(use_with daemon) \
+		$(use_with dbus qtdbus) \
+		$(use_with qt5 gui qt5) \
+		$(use_with qrcode qrencode) \
+		$(use_with upnp miniupnpc) \
+		$(use_with pic) \
+		$(use_with libraries libs) \
+		$(use_with utils) \
+		$(use_enable harden hardening) \
+		$(use_enable bench) \
+		$(use_enable ccache ) \
+		$(use_enable static) \
+		$(use_enable debug) \
+		$(use_enable test tests)
+}
+
+src_install() {
+	# Live package is called staging and should *only* be used for testing purposes
+	if [[ ${PV} == 9999 ]]
+	then
+		if use daemon
+		then
+			newbin src/gridcoinresearchd gridcoinresearchd-staging
+			newman doc/gridcoinresearchd.1 gridcoinresearchd-staging.1
+			newinitd "${FILESDIR}"/gridcoin-staging.init gridcoin-staging
+			systemd_dounit "${FILESDIR}"/gridcoin-staging.service
+		fi
+		if use qt5
+		then
+			newbin src/qt/gridcoinresearch gridcoinresearch-staging
+			newman doc/gridcoinresearch.1 gridcoinresearch-staging.1
+			newmenu contrib/gridcoinresearch.desktop gridcoinresearch-staging.desktop
+			for size in 16 22 24 32 48 64 128 256 ; do
+				newicon -s "${size}" "share/icons/hicolor/${size}x${size}/apps/gridcoinresearch.png" gridcoinresearch-staging.png
+			done
+			newicon -s scalable "share/icons/hicolor/scalable/apps/gridcoinresearch.svg" gridcoinresearch-staging.svg
+		fi
+	else
+		if use daemon
+		then
+			newbin src/gridcoinresearchd gridcoinresearchd
+			newman doc/gridcoinresearchd.1 gridcoinresearchd.1
+			newinitd "${FILESDIR}"/gridcoin.init gridcoin
+			systemd_dounit "${FILESDIR}"/gridcoin.service
+		fi
+		if use qt5
+		then
+			newbin src/qt/gridcoinresearch gridcoinresearch
+			newman doc/gridcoinresearch.1 gridcoinresearch.1
+			domenu contrib/gridcoinresearch.desktop
+			for size in 16 22 24 32 48 64 128 256 ; do
+				doicon -s "${size}" "share/icons/hicolor/${size}x${size}/apps/gridcoinresearch.png"
+			done
+			doicon -s scalable "share/icons/hicolor/scalable/apps/gridcoinresearch.svg"
+		fi
+	fi
+
+	dodoc README.md CHANGELOG.md doc/build-unix.md
+
+	diropts -o${PN} -g${PN}
+	keepdir /var/lib/${PN}/.GridcoinResearch/
+	newconfd "${FILESDIR}"/gridcoinresearch.conf gridcoinresearch
+	fowners gridcoin:gridcoin /etc/conf.d/gridcoinresearch
+	fperms 0660 /etc/conf.d/gridcoinresearch
+	dosym ../../../../../etc/conf.d/gridcoinresearch /var/lib/${PN}/.GridcoinResearch/gridcoinresearch.conf
+}
+
+pkg_postinst() {
+	elog "You are using a source compiled version of gridcoin."
```

Comment:
I'm not sure this is worth saying. What's the point it's getting at? Does it hurt upstream support somehow?